### PR TITLE
Branches -- Paige

### PR DESCRIPTION
**Pull request by minipaige02**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | The Model is the part in Rails that is communicating with the database and making adjustments to it as needed.
Describe in your own words what the Controller is doing in Rails | The Controller is processing and responding to events that it is receiving from the router. It is also passing information between the Model, View, and the Browser.
Describe in your own words what the View is doing in Rails | The View takes data from the Controller, turns it into HTML (adds any CSS) and sends it back to the Controller to send it on to the Browser.
Describe an edge-case controller test you wrote | I wrote a test for the 'destroy' action that made sure that the database was not changed if you tried to delete a task that had already been deleted.
What is the purpose of using strong params? (i.e. the params method in the controller) | Using strong params is a safe way to require and permit the data that is coming into forms that we have set up in our application. It also saves us time when we need to update the schema to our database or make other changes to the application that involves params.
How are Rails migrations related to Rails models? | Rails migrations make changes to the model's database table. 
Describe one area of Rails that are still unclear on | I'm still a little unclear on where the database that Rails creates on our machine is stored. Does Rails save something that points to the database or is the database somewhere within the application that never gets uploaded to GitHub when we push our code to the site?

### CODE REVIEW

**Comment by tildeee** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | x
Answered comprehension questions | x
Successfully handles: Index, Show | x
Index & Show tests pass | x
Successfully handles: New, Create | x
New & Create tests pass | x
Successfully handles: Edit, Update | x
Tests for Edit & Update test valid & invalid task ids | x
Successfully handles: Destroy, Task Complete | x
Tests for Destroy & Task Complete include tests for valid and invalid task ids | x
Routes follow RESTful conventions | x
Uses named routes (like `_path`) | spots for improvements in tests
**Overall** |



Well done on this project, Paige! Your app code and test code is clean, readable, and working just as expected.


I have a comment that your tests sometimes referred to a URL with the pattern `/tasks/some_id` etc, rather than using named paths. In the future, it'll be our best practice to use named paths.


That being said, your code looks great! Keep up the great work!

File: test/controllers/tasks_controller_test.rb [hunk 1]
```diff
@@ -79,48 +72,172 @@
...
+      updated_task_data = {
+        task: {
+          name: "Walk dog",
+          description: "Walk the dog",
+          completion_date: nil,
+        },
+      }
+      task_url = "/tasks/" + "#{task.id}"
+      
+      patch task_url, params: updated_task_data
```
Comment (tildeee) [inline test/controllers/tasks_controller_test.rb:109]: Instead of task_url, hopefully in the future I'd like to see this use the named path (in this case, it'd be `task_path`)

File: test/controllers/tasks_controller_test.rb [hunk 2]
```diff
@@ -79,48 +72,172 @@
...
+      
+      expect(Task.count).must_equal 1
+      must_redirect_to task_path(id: task.id)
+      expect(Task.find_by(id: task.id).name).must_equal "Walk dog"
     end
-
+    
     it "will redirect to the root page if given an invalid id" do
-      # Your code here
+      patch '/tasks/0', params: {
```
Comment (tildeee) [inline test/controllers/tasks_controller_test.rb:117]: Same as above-- try `patch task_path 0, params: { ... }`

File: test/controllers/tasks_controller_test.rb [hunk 3]
```diff
@@ -79,48 +72,172 @@
...
+      }.must_differ "Task.count", 0
 
+      must_redirect_to tasks_path
+    end
   end
 
   # Complete for Wave 4
   describe "toggle_complete" do
-    # Your tests go here
+    it "successfully updates completion_date to today's date if nil and redirects to the tasks path" do
```
Comment (tildeee) [inline test/controllers/tasks_controller_test.rb:196]: Nicely done on all of these `toggle_complete` tests! They are thorough and well-tested!

File: db/seeds.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  {name: "Go to Brunch", description: "Eats lots of yummy food and drink coffee"},
+  {name: "Go to Lunch", description: "Eat more food"},
+  {name: "Go to Second Lunch", description: "Eat a snack. Maybe take a nap."},
+  {name: "Play Video Games", description: "Download Untitled Gooose Game", completion_date: Date.today},
+  {name: "High Five Somebody You Don't Know", description: "Use hand sanitizer after. It's cold season", completion_date: Date.today},
+  {name: "Plant Flowers", description: "Get some pots first.", completion_date: Date.today},
+  {name: "Call Mom", description: "She worries, you know."},
+  {name: "Nap.", description: "Period.", completion_date: Date.today}])
+
+  puts "Total Tasks added: #{Task.count} / 9"
```
Comment (tildeee) [inline db/seeds.rb:20]: Fantastic seed script :D
